src/quote/leveraged-extended/provider.ts

```diff
@@ -0,0 +1,217 @@
+import { BigNumber } from '@ethersproject/bignumber'
+import { JsonRpcProvider } from '@ethersproject/providers'
+
+import { ETH, InterestCompoundingETHIndex } from 'constants/tokens'
+import { ZeroExApi } from 'utils/0x'
+import { slippageAdjustedTokenAmount } from 'utils/slippage'
+import {
+  Exchange,
+  getSwapDataCollateralDebt,
+  getSwapDataDebtCollateral,
+  getSwapData,
+  SwapData,
+} from 'utils/swapData'
+
+import { QuoteProvider } from '../quoteProvider'
+import { QuoteToken } from '../quoteToken'
+
+import { getLeveragedTokenData } from './utils/data'
+import { getIncludedSources } from './utils/zeroex'
+
+export interface FlashMintLeveragedExtendedQuoteRequest {
+  isMinting: boolean
+  inputToken: QuoteToken
+  outputToken: QuoteToken
+  indexTokenAmount: BigNumber
+  slippage: number
+}
+
+export interface FlashMintLeveragedExtendedQuote {
+  indexTokenAmount: BigNumber
+  inputOutputTokenAmount: BigNumber
+  swapDataDebtCollateral: SwapData
+  swapDataPaymentToken: SwapData
+}
+
+export interface LeveragedTokenData {
+  collateralAToken: string
+  collateralToken: string
+  debtToken: string
+  collateralAmount: BigNumber
+  debtAmount: BigNumber
+}
+
+export class LeveragedExtendedQuoteProvider
+  implements
+    QuoteProvider<
+      FlashMintLeveragedExtendedQuoteRequest,
+      FlashMintLeveragedExtendedQuote
+    >
+{
+  constructor(
+    private readonly provider: JsonRpcProvider,
+    private readonly zeroExApi: ZeroExApi
+  ) {}
+
+  async getQuote(
+    request: FlashMintLeveragedExtendedQuoteRequest
+  ): Promise<FlashMintLeveragedExtendedQuote | null> {
+    const { provider, zeroExApi } = this
+    const { inputToken, indexTokenAmount, isMinting, outputToken, slippage } =
+      request
+    const indexToken = isMinting ? outputToken : inputToken
+    const indexTokenSymbol = indexToken.symbol
+    const includedSources = getIncludedSources()
+    const network = await provider.getNetwork()
+    const chainId = network.chainId
+    console.log('chainId:', chainId)
+
+    const leveragedTokenData = await getLeveragedTokenData(
+      indexToken.address,
+      indexTokenAmount,
+      indexTokenSymbol,
+      isMinting,
+      chainId,
+      provider
+    )
+    console.log(leveragedTokenData)
```

Comment:
Biggest question right now is if we now instead of `indexTokenAmount` always just have the `inputTokenAmount` but [getLeveragedTokenData](https://arbiscan.io/address/0xC62e39d1f5232f154b7ccD3C6234A9c893bf9563#readContract#F7) only takes the setTokenAmount (indexTokenAmount) does this somehow influence the calculations.

In general, redemptions should work - as nothing has changed here. We're still using the indexTokenAmount (which is the inputTokenAmount in case of redeeming) and can return a quote.

But for issuance the question is what needs to change in here to calc the quote? @0xonramp @ckoopmann